include/onnxruntime/core/framework/ortmemoryinfo.h

```diff
@@ -41,7 +41,7 @@ struct OrtMemoryInfo {
   std::string ToString() const {
     std::ostringstream ostr;
     ostr << "OrtMemoryInfo:["
-         << "name:" << name
+         << "name:" << (name ? name : "")
```

Comment:
why we have case that the name is a nullptr?

Reply:
If someone uses the default constructor then name can be nullptr. I feel if the struct is providing a default constructor or a case where the name can be nullptr then ideally all the member function should check for null safety.